src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -1001,7 +1001,7 @@ private static string GenerateFastPathFuncForType(TypeGenerationSpec typeGenSpec
 
                 return $@"
 
-private static void {serializeMethodName}({Utf8JsonWriterTypeRef} {WriterVarName}, {valueTypeRef} {ValueVarName})
+private void {serializeMethodName}({Utf8JsonWriterTypeRef} {WriterVarName}, {valueTypeRef} {ValueVarName})
```

Comment:
I'm ok with this change but this seems counter-intuitive, is there any work being done to make static faster? Can we at least add a comment why this is not static so that we don't accidentally refactor it back

Reply:
> is there any work being done to make static faster

It cannot be done.

Invoking a delegate == calling `someDelegate.Invoke`. At that point we don't know what method the delegate points to. Codegen needs to decide how to call it (lay out the arguments) so it assumes "instance method". It loads `this` loads all arguments and calls whatever function pointer the delegate has. If this was an instance method, we're good to go. If this wasn't an instance method the function pointer stored within the delegate is actually a "shuffle thunk" that removes the first parameter (now unused `this`) and moves all parameter where the method expects them. _Then_ it calls the method.

We need to optimize for _some_ case and the optimized case is instance methods. Roslyn also hardcodes this (lambda captures are instance methods on a dummy object instance even though they could be static).